openpype/modules/kitsu/utils/update_op_with_zou.py

```diff
@@ -0,0 +1,376 @@
+"""Functions to update OpenPype data using Kitsu DB (a.k.a Zou)."""
+from copy import deepcopy
+from typing import Dict, List
+
+from pymongo import DeleteOne, UpdateOne
+from pymongo.collection import Collection
+import gazu
+from gazu.task import (
+    all_tasks_for_asset,
+    all_tasks_for_shot,
+)
+
+from openpype.pipeline import AvalonMongoDB
+from openpype.api import get_project_settings
+from openpype.lib import create_project
+from openpype.modules.kitsu.utils.credentials import validate_credentials
+
+
+def create_op_asset(gazu_entity: dict) -> dict:
+    """Create OP asset dict from gazu entity.
+
+    :param gazu_entity:
+    """
+    return {
+        "name": gazu_entity["name"],
+        "type": "asset",
+        "schema": "openpype:asset-3.0",
+        "data": {"zou": gazu_entity, "tasks": {}},
+    }
+
+
+def set_op_project(dbcon, project_id) -> Collection:
+    """Set project context.
+
+    :param dbcon: Connection to DB.
+    :param project_id: Project zou ID
+    """
+    project = gazu.project.get_project(project_id)
+    project_name = project["name"]
+    dbcon.Session["AVALON_PROJECT"] = project_name
+
+    return dbcon.database[project_name]
```

Comment:
You change `AVALON_PROJECT`, which affect how `dbcon` object behaves, then you don't use the changed behavior but return direct access to project's database instead. So the change of `AVALON_PROJECT` is pointless.

NOTE: Ideally, you should not use direct access to database but through object of `AvalonMongoDB`. If you want make sure that project is not changed, then create new object of `AvalonMongoDB`, they share mongo connection.
```
# This is how it was meant to use 'AvalonMongoDB'
dbcon = AvalonMongoDB()

dbcon.Session["AVALON_PROJECT"] = project_name
dbcon.find_one(...)
dbcon.update_one(...)

# This is how you use it
dbcon = AvalonMongoDB()

col = dbcon.database[project_name]
col.find_one(...)
col.update_one(...)
```

Reply:
You mean it'd be better to not directly use `col` but `Session[]`?

To me it seems cleaner to access directly the database collection instead of using some general session setting no?

Reply:
> You mean it'd be better to not directly use col but Session[]?

You should use methods on object of `AvalonMongoDB` instead of direct access of project collection.

> To me it seems cleaner to access directly the database collection instead of using some general session setting no?

It's not general. Each `AvalonMongoDB` object has it's own session. Only general session is  `openpype.pipeline.legacy_io.Session`.

This would work exactly the same but using AvalonMongoDB object:
```
def set_op_project(dbcon, project_id) -> AvalonMongoDB:
    """Set project context.
    :param dbcon: Connection to DB.
    :param project_id: Project zou ID
    """
    project = gazu.project.get_project(project_id)
    project_name = project["name"]
    dbcon.Session["AVALON_PROJECT"] = project_name
    return dbcon
```

And this would be "thread safe" for cases that multiple threads would access the same AvalonMongoDB object for different projects.
```
def get_project_con(project_id) -> Collection:
    project = gazu.project.get_project(project_id)
    project_name = project["name"]
    dbcon = AvalonMongoDB()
    dbcon.Session["AVALON_PROJECT"] = project_name
    return dbcon
```

Reply:
I changed everything, could you tell me if it fits your expectations?